tests/dotnet/UnitTests/ProjectTest.cs

```diff
@@ -682,6 +686,205 @@ public void AppWithResources (ApplePlatform platform, string runtimeIdentifiers)
+			DirectoryInfo xcassetsDir = new DirectoryInfo (Path.Combine (project_path, @"../Assets.xcassets"));
+
+			Assert.That (testingAssetsDir, Does.Exist);
+			if (!Directory.Exists (xcassetsDir.FullName))
+				MakeSymlinks (testingAssetsDir.FullName, xcassetsDir.FullName);
+			Assert.That (xcassetsDir, Does.Exist);
+		}
+
+		void MakeSymlinks (string sourceDir, string destDir)
+		{
+			var output = new StringBuilder ();
+			var executable = "ln";
+			var arguments = new string [] { "-s", sourceDir, destDir };
+			var rv = Execution.RunWithStringBuildersAsync (executable, arguments, standardOutput: output, standardError: output, timeout: TimeSpan.FromSeconds (60)).Result;
+			Assert.AreEqual (0, rv.ExitCode, "ExitCode");
+			return;
+		}
+
+		string GetFullSdkVersion (string sdkVersion) => sdkVersion switch {
+			"iphonesimulator" => sdkVersion + Configuration.sdk_version,
+			"iphoneos" => sdkVersion + Configuration.sdk_version,
+			"appletvsimulator" => sdkVersion + Configuration.tvos_sdk_version,
+			"macosx" => sdkVersion + Configuration.macos_sdk_version,
+			_ => throw new ArgumentOutOfRangeException (nameof (sdkVersion), $"Not expected sdkVersion: {sdkVersion}"),
+		};
+
+		JsonDocument ProcessAssets (string assetsPath, string sdkVersion) {
+			var output = new StringBuilder ();
+			var executable = "xcrun";
+			var arguments = new string [] { "--sdk", sdkVersion, "assetutil", "--info", assetsPath };
+			var rv = Execution.RunWithStringBuildersAsync (executable, arguments, standardOutput: output, standardError: output, timeout: TimeSpan.FromSeconds (120)).Result;
+			Assert.AreEqual (0, rv.ExitCode, "ExitCode");
+			var s = output.ToString ();
+
+			// This Execution call produces an output with an objc warning. We just want the json below it.
+			if (s.StartsWith ("objc"))
+				output.Remove (0, s.IndexOf (Environment.NewLine) + 1);
+
+			return JsonDocument.Parse (output.ToString ());
+		}
+
+		HashSet<string> FindAssets (JsonDocument doc)
+		{
+			var jsonArray = doc.RootElement.EnumerateArray ();
+			var foundElements = new HashSet<string> ();
+
+			foreach (var item in jsonArray) {
+				var result = GetTarget (item);
+				if (result is not null)
+					foundElements.Add (result);
+			}
+			return foundElements;
+		}
+
+		string? GetTarget (JsonElement item) {
+			if (item.TryGetProperty ("AssetType", out var assetType)) {
+				foreach (var target in XCAssetTargets) {
+					var result = GetTarget (item, assetType, target);
+					if (result is not null)
+						return result;
+				}
+			}
+			return null;
+		}
+
+		string? GetTarget (JsonElement item, JsonElement assetType, XCAssetTarget target)
+		{
+			if (assetType.ToString () == target.AssetType) {
+				if (item.TryGetProperty (target.CategoryName, out var value)) {
+					if (target.Values.Contains (value.ToString ()))
+						return string.Concat (assetType.ToString (), ".", value.ToString ());
+				}
+			}
+			return null;
+		}
+
+		class XCAssetTarget {
```

Comment:
could we use a struct?

Reply:
Here's a snippet of what Microsoft docs say:
```
❌ AVOID defining a struct unless the type has all of the following characteristics:

- It logically represents a single value, similar to primitive types (int, double, etc.).
- It has an instance size under 16 bytes.
- It is immutable.
- It will not have to be boxed frequently.

In all other cases, you should define your types as classes.
```
Since there are string arrays inside the XCAssetTarget, I am thinking that the instance size will be much greater than 16 bytes in most of the uses if I understand that correctly. Therefore I am leaning towards a class?

Reply:
Not of terrible importance to switch to a struct perse but you are not correctly understanding how the array is stored. Please read: https://stackoverflow.com/questions/1113819/arrays-heap-and-stack-and-value-types and pay attention at some nice graphics at the bottom of the post.

Reply:
When don't care that much in the tests, leave as it is ;)